File: API/Entities/WeatherStation.cs
using CsvHelper.Configuration.Attributes;

namespace API.Entities;

public class WeatherStation
{
    public int Id { get; set; }
    public string Name { get; set; }
    public string Site { get; set; }
    public string Portfolio { get; set; }
    public string State { get; set; }
    public double Latitude { get; set; }
    public double Longitude { get; set; }
    public ICollection<WeatherVariable> Variables { get; set; }
}

File: client/src/app/models/weatherStation.ts
export interface VariableData {
  id: string;
  timestamp: string;
  value: number;
}

export interface Variable {
  id: number;
  name: string;
  unit: string;
  longName: string;
  data: VariableData[];
}

export interface weatherStation {
  id: number;
  name: string;
  site: string;
  portfolio: string;
  state: string;
  latitude: number;
  longitude: number;
  variables: Variable[];
}
